crates/executor/src/execution_state.rs

```diff
@@ -57,22 +65,101 @@ impl<'tx> ExecutionState<'tx> {
+        );
+
+        let chain_id: Vec<_> = self
+            .chain_id
+            .0
+            .to_be_bytes()
+            .into_iter()
+            .skip_while(|b| *b == 0)
+            .collect();
+        let chain_id = String::from_utf8(chain_id)?;
+
+        Ok(ChainInfo {
+            chain_id: starknet_api::core::ChainId(chain_id),
+            fee_token_addresses: blockifier::context::FeeTokenAddresses {
+                strk_fee_token_address,
+                eth_fee_token_address,
+            },
+        })
+    }
+
+    fn block_info(&self) -> anyhow::Result<BlockInfo> {
+        Ok(BlockInfo {
+            block_number: starknet_api::block::BlockNumber(self.header.number.get()),
+            block_timestamp: starknet_api::block::BlockTimestamp(self.header.timestamp.get()),
+            sequencer_address: starknet_api::core::ContractAddress(
+                PatriciaKey::try_from(self.header.sequencer_address.0.into_starkfelt())
+                    .expect("Sequencer address overflow"),
+            ),
+            gas_prices: blockifier::block::GasPrices {
+                eth_l1_gas_price: if self.header.eth_l1_gas_price.0 == 0 {
+                    // Bad API design - the genesis block has 0 gas price, but
+                    // blockifier doesn't allow for it. This isn't critical for
+                    // consensus, so we just use 1.
+                    1.try_into().unwrap()
+                } else {
+                    self.header.eth_l1_gas_price.0.try_into().unwrap()
+                },
+                strk_l1_gas_price: if self.header.strk_l1_gas_price.0 == 0 {
+                    // Bad API design - the genesis block has 0 gas price, but
+                    // blockifier doesn't allow for it. This isn't critical for
+                    // consensus, so we just use 1.
+                    1.try_into().unwrap()
+                } else {
+                    self.header.strk_l1_gas_price.0.try_into().unwrap()
+                },
+                eth_l1_data_gas_price: if self.header.eth_l1_data_gas_price.0 == 0 {
+                    // Bad API design - pre-v0.13.1 blocks have 0 data gas price, but
+                    // blockifier doesn't allow for it. This value is ignored for those
+                    // transactions.
+                    1.try_into().unwrap()
+                } else {
+                    self.header.eth_l1_data_gas_price.0.try_into().unwrap()
+                },
+                strk_l1_data_gas_price: if self.header.strk_l1_data_gas_price.0 == 0 {
+                    // Bad API design - pre-v0.13.1 blocks have 0 data gas price, but
+                    // blockifier doesn't allow for it. This value is ignored for those
+                    // transactions.
+                    1.try_into().unwrap()
+                } else {
+                    self.header.strk_l1_data_gas_price.0.try_into().unwrap()
+                },
+            },
+            use_kzg_da: false,
```

Comment:
Ah, this _will_ have to get fixed otherwise blockifier won't correctly use data gas.

We'll have to use either true/false depending on the `l1_da_mode` field of the block header. I guess we'll need to add that to the `pathfinder_common::BlockHeader` type too.